### PR TITLE
reworked taar_lite job to use moz_dataproc_pyspark_runner

### PR DESCRIPTION
**Pull request by crankycoder**
WIP:

Preliminary port of taar-lite job to run in GCP

### CODE REVIEW

**Comment by crankycoder** [comment]
@haroldwoo @akkomar   I'm not sure where I should be putting my taar-lite script in GCS.   I've commented it on line https://github.com/mozilla/telemetry-airflow/pull/608/files#diff-b7ffb575ceeae29c0d825844f691b326R105

File: dags/taar_amodump.py [hunk 1]
```diff
@@ -2,21 +2,47 @@
...
+from airflow.contrib.hooks.aws_hook import AwsHook
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
 
 from utils.mozetl import mozetl_envvar
+from utils.dataproc import moz_dataproc_pyspark_runner
+
+# Unsalted cluster name so subsequent runs fail if the cluster name exists
+cluster_name = "taarlite-dataproc-cluster"
+
+# We use an application-specific gcs bucket since the copy operator can't set the destination
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: We probably don't need this. What is the output of this dag meant to be?

File: dags/taar_amodump.py [hunk 2]
```diff
@@ -2,21 +2,47 @@
...
+# of having the dataset name come first
+gcs_data_bucket = "moz-fx-data-prod-taarlite-data"
+
+dataset = "taarlite_guidranking"
+
+# Defined in Airflow's UI -> Admin -> Connections
+gcp_conn_id = "google_cloud_airflow_dataproc"
+connection = GoogleCloudBaseHook(gcp_conn_id=gcp_conn_id)
+
+# Required to copy socorro json data from aws prod s3 to gcs
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: prob not needed either, what is the input data?

File: dags/taar_amodump.py [hunk 3]
```diff
@@ -2,21 +2,47 @@
...
+read_aws_conn_id = "aws_taarlite_readonly_s3"
+
+# Required to write parquet data back to s3://telemetry-parquet
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: If we're writing back to aws we'd need an aws connection here

File: dags/taar_amodump.py [hunk 4]
```diff
@@ -2,21 +2,47 @@
...
 default_args = {
-    'owner': 'amiyaguchi@mozilla.com',
-    'depends_on_past': False,
-    'start_date': datetime(2018, 11, 26),
-    'email': ['telemetry-alerts@mozilla.com', 'amiyaguchi@mozilla.com'],
-    'email_on_failure': True,
-    'email_on_retry': True,
-    'retries': 3,
-    'retry_delay': timedelta(minutes=30),
+    "owner": "amiyaguchi@mozilla.com",
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: change to vng, hwoo?

File: dags/taar_amodump.py [hunk 5]
```diff
@@ -2,21 +2,47 @@
...
+    "depends_on_past": False,
+    "start_date": datetime(2018, 11, 26),
+    "email": ["telemetry-alerts@mozilla.com", "amiyaguchi@mozilla.com"],
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: same here

File: dags/taar_amodump.py [hunk 6]
```diff
@@ -55,27 +81,37 @@
...
-    uri="https://raw.githubusercontent.com/mozilla/python_mozetl/master/bin/mozetl-submit.sh",
-    output_visibility="private",
-    dag=dag
+    dag=dag,
+    subdag=moz_dataproc_pyspark_runner(
+        parent_dag_name=dag.dag_id,
+        default_args=default_args,
+        cluster_name=cluster_name,
+        job_name="Generate GUID coinstallation JSON for TAAR",
+        # TODO - CI/CD for gcs and repo sync for artifact bucket jobs and gcp bootstrap folders
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: can rm these comments, this is done now

File: dags/taar_amodump.py [hunk 7]
```diff
@@ -55,27 +81,37 @@
...
+        # TODO: the currentcode lives at - need to get it into GCS
+        # https://github.com/mozilla/python_mozetl/blob/master/mozetl/taar/taar_lite_guidguid.py
+        python_driver_code="gs://moz-fx-data-prod-airflow-dataproc-artifacts/jobs/socorro_import_crash_data.py",
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: this path should point to the same gcs bucket, but then a file in telemetry-airflow/jobs  (which will be sync'd to that bucket on master commits)

File: dags/taar_amodump.py [hunk 8]
```diff
@@ -55,27 +81,37 @@
...
+        py_args=["--date", "{{ ds_nodash }}"],
```
Comment (haroldwoo) [inline dags/taar_amodump.py]: does your script expect --date?

File: dags/taar_amodump.py (hunk 7 above)
Comment (haroldwoo) [reply dags/taar_amodump.py]: For testing purposes, i can drop the script there for you manually

**Comment by crankycoder** [comment]
I've reworked a lot of this with @hwoo.  I'll open a new PR.

**Pull request closed without merging**